src/domain/kaivuilmoitus/components/HaittojenhallintaSuunnitelma.tsx

```diff
@@ -25,19 +28,69 @@ import styles from './HaittojenhallintaSuunnitelma.module.scss';
 import HaittojenhallintaMap from './HaittojenhallintaMap';
 import useIsHaittojenhallintaSectionVisible from '../../common/haittojenhallinta/useIsHaittojenhallintaSectionVisible';
 import ProcedureTips from '../../common/haittaIndexes/ProcedureTips';
+import { HaittaIndexData } from '../../common/haittaIndexes/types';
 
 type Props = {
   hankeAlue: HankeAlue;
   kaivuilmoitusAlue: KaivuilmoitusAlue;
   index: number;
 };
 
+type HankeNuisanceControlProps = {
+  indeksi?: number;
+  haitta: HAITTOJENHALLINTATYYPPI;
+  haittojenhallintasuunnitelma?: Haittojenhallintasuunnitelma;
+};
+
+function HankeNuisanceControl({
+  indeksi = 0,
+  haitta,
+  haittojenhallintasuunnitelma,
+}: Readonly<HankeNuisanceControlProps>) {
+  const { t } = useTranslation();
+
+  if (!haitta) return null;
+
+  const hankeText = haittojenhallintasuunnitelma?.[haitta];
+
+  return (
+    <Box mb="var(--spacing-m)">
+      <Notification
+        label={t('hakemus:labels:hankeAreaNuisanceControl')}
+        style={{ overflow: 'visible' }}
+      >
+        {haitta !== HAITTOJENHALLINTATYYPPI.YLEINEN && haitta !== HAITTOJENHALLINTATYYPPI.MUUT && (
+          <Box mb="var(--spacing-s)">
+            <HaittaIndexHeading
+              index={indeksi}
+              haittojenhallintaTyyppi={haitta}
+              heading={t('kaivuilmoitusForm:haittojenHallinta:hankeHaittaindeksi')}
+              testId={`test-hanke-${haitta}`}
+            />
+          </Box>
+        )}
+        <Box as="p" className="text-sm" data-testid={`test-hanke-nuisance-control-${haitta}`}>
```

Comment:
`whiteSpace="pre-wrap"` should be added here to preserve line breaks in the text.

Reply:
Added white-space rule.